### DISCUSSION TITLE
Automatic import .targets from referenced project

### DISCUSSION THREAD

**Discussion by kefyru** [dotnet/msbuild#8381] [Q&A]
Two projects:
SomeLib.csproj
SomeApp.csproj (uses SomeLib)

1. SomeLib/bin/SomeLib.csproj.*.targets automatically applies to SomeLib project
2. SomeLib/build/SomeLib.targets automatically applies to dependent projects via `<PackageReference Include="SomeLib" />`
3. Directory.build.targets automatically applies to all projects in this directory (if not redefined in project dir)

Q: Is there any possibility to make some .targets file which will be applied automatically to SomeApp via `<ProjectReference Include="SomeLib.csproj"/>`

For example
```xml
<!-- SomeLib/??/??.targets -->
<Project>
  <Target Name="SomeLibAutoTarget" BeforeTargets="Build">
    <Message Text="Hello '$(ProjectName)'"/>
  </Target>
</Project>
```

```xml
<!-- SomeApp.csproj -->
<Project>
  <ItemGroup>
    <ProjectReference Include="..\SomeLib\SomeLib.csproj"/>
  </ItemGroup>
</Project>
```

`SomeApp % dotnet build`
--> Hello 'SomeApp'

**Comment by JanKrivanek** [comment]
The suggested way is to extract the desired targets/props into a separate file and include it into projects explicitly. If the downstream consumers are independent from the project of question - then the reference should vbe packaged and distributed as nuget package (which supports auto-importable build scripts).

If you have scenario that is no achievable via any of the above options, then please provide a bit more info on the actual intent

**Reply by baronfel** [reply]
This is also IMO an aspect of Project/Package substitution - users get this behavior for Packages and so they would also like it for Projects. This makes sense to me, it's just hard to do correctly for all cases.
